packages/grpc-native-core/src/client.js

```diff
@@ -53,6 +53,21 @@ var Duplex = stream.Duplex;
 var util = require('util');
 var version = require('../package.json').version;
 
+/**
+ * Create an Error object from a status object
+ * @private
+ * @param {grpc~StatusObject} status The status object
+ * @return {Error} The resulting Error
+ */
+function createStatusError(status) {
+  let statusName = _.invert(constants.status)[status.code];
+  let message = `${status.code} ${status.name}: ${status.details}`;
+  let error = new Error(message);
+  error.code = status.code;
+  error.details = status.details;
```

Comment:
Should we include `error.metadata` as well?

Reply:
Yes. That is actually what I meant to write there. The details string is in the message. It probably doesn't need to be in a separate field too.